### PR TITLE
Use RJSF schema from Sistent in Meshery UI

### LINKED ISSUE #10445: Use RJSF schema from Sistent in Meshery UI
### Mockups

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by 11happy**
**Notes for Reviewers**

This PR fixes #10445

**Components from which API is removed:**
- `ui/components/Modals/PublishModal.js`
- `ui/components/MesheryPatterns.js`
- `ui/components/Filters.js`

**Acceptance test:**
- [x] Remove existing API calls for getting schemas
- [x] Use Sistent library to import schemas
- [x] No change should be made in UI or where ever we are using schemas, if required give reason why
- [x] Add checklist of components from which API is removed
- [x] Add checklist of all components which using API to fetch schema
 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 
**CC:**
- @Yashsharma1911

### CODE REVIEW

**Comment by 11happy** [comment]
Hello @yash37158 can you please take a look, I have currently removed the api related code.

File: ui/components/Filters.js [hunk 1]
```diff
@@ -334,44 +335,6 @@ function MesheryFilters({
...
-          modelNames.sort();
-
-          // Modify the schema using the utility function
-          const modifiedSchema = modifyRJSFSchema(
-            result.rjsfSchema,
-            'properties.compatibility.items.enum',
-            modelNames,
-          );
-          setPublishSchema({ rjsfSchema: modifiedSchema, uiSchema: result.uiSchema });
-          setMeshModels(models);
```
Comment (Yashsharma1911) [inline ui/components/Filters.js:367]: @11happy thanks for removing this code, also it seems like we also removed code for "setMeshModels" here, by default publish schema doesn't have meshmodels, for that we have this function where we fetch meshmodels and update schema `properties.compatibility.items.enum` property with those meshmodels, as we are now fetching it directly from sistent this will cause issue and publish modal will not show any dropdown in technology field becuse meshmodels is not present in publish schema

Can you make sure for publish schema we update meshmodels, you can use this function to update meshmodels to schema property

```

          // Modify the schema using the utility function
          const modifiedSchema = modifyRJSFSchema(
            result.rjsfSchema,
            'properties.compatibility.items.enum',
            modelNames,
          );
```

As you can see in below shema link, in this enum field only kubernetes is present by default, however there should be 190 meshmodels, and to make sure they are dynamic in nature and we don't need to update schema again and again we update publish schema in Meshery UI, so can you make sure that we don't remove that logic.

https://github.com/layer5io/sistent/blob/master/src/schemas/publishCatalogItem/schema.tsx#L12

You can have a test case that

- [ ] Changes made in migrating schema from API to Sistent in Meshery UI should not affect the UI

File: ui/components/MesheryPatterns.js [hunk 2]
```diff
@@ -594,45 +594,6 @@ function MesheryPatterns({
...
-
-          // Modify the schema using the utility function
-          const modifiedSchema = modifyRJSFSchema(
-            result.rjsfSchema,
-            'properties.compatibility.items.enum',
-            modelNames,
-          );
-
-          setPublishSchema({ rjsfSchema: modifiedSchema, uiSchema: result.uiSchema });
-          setMeshModels(models);
```
Comment (Yashsharma1911) [inline ui/components/MesheryPatterns.js:628]: Same here

File: ui/components/Modals/PublishModal.js [hunk 3]
```diff
@@ -1,50 +1,16 @@
...
-          modelNames.sort();
-
-          // Modify the schema using the utility function
-          const modifiedSchema = modifyRJSFSchema(
-            result.rjsfSchema,
-            'properties.compatibility.items.enum',
-            modelNames,
-          );
-
-          setPublishSchema({ rjsfSchema: modifiedSchema, uiSchema: result.uiSchema });
```
Comment (Yashsharma1911) [inline ui/components/Modals/PublishModal.js:34]: Same here

**Review by Yashsharma1911 — CHANGES_REQUESTED** [review]
Thanks for picking this up and raising PR @11happy

**Comment by vishalvivekm** [comment]
@11happy Let's discuss this in Meshery dev call today. Add this as an agenda item to the meeting[ minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#), if you would :)

**Comment by Yashsharma1911** [comment]
Uhh ho! merge conflict

**Review by Yashsharma1911 — COMMENTED** [review]
Please make sure you add these test cases marked in your PR description

- [ ] Remove existing API calls for getting schemas
- [ ] Use Sistent library to import schemas
- [ ] No change should be made in UI or where ever we are using schemas, if required give reason why
- [ ] Add checklist of components from which API is removed
- [ ] Add checklist of all components which using API to fetch schema

**Comment by 11happy** [comment]
I have taken reference from the @sudhanshutech & also included his work for notify & other things.

File: ui/components/Filters.js (hunk 1 above)
Comment (11happy) [reply ui/components/Filters.js:369]: done

File: ui/components/MesheryPatterns.js (hunk 2 above)
Comment (11happy) [reply ui/components/MesheryPatterns.js]: done

File: ui/components/Modals/PublishModal.js (hunk 3 above)
Comment (11happy) [reply ui/components/Modals/PublishModal.js]: done

**Comment by 11happy** [comment]
> Uhh ho! merge conflict

removed the merge conflict as well.

File: ui/components/Filters.js [hunk 4]
```diff
@@ -1470,13 +1462,13 @@ const ImportModal = React.memo((props) => {
...
-  const { publishFormSchema, handleClose, handlePublish, title } = props;
+  const { handleClose, handlePublish, title } = props;
 
   return (
     <Modal
       open={true}
-      schema={publishFormSchema.rjsfSchema}
-      uiSchema={publishFormSchema.uiSchema}
+      schema={publishCatalogItemSchema}
+      uiSchema={publishCatalogItemUiSchema}
```
Comment (Yashsharma1911) [inline ui/components/Filters.js:1471]: Thankyou for updating @sudhanshutech 's PR changes too, however here do make sure you are using updating schema that you updated [here](https://github.com/meshery/meshery/pull/10720/files#diff-6ec8e150ffa5db5b0830cd1598d529bbc98c79c98df32092858a51a779fcdea0R355)
```suggestion
      schema={publishFormSchema.rjsfSchema}
      uiSchema={publishFormSchema.uiSchema}
```

**Review by Yashsharma1911 — CHANGES_REQUESTED** [review]
Thanks after this change it looks good to me, also after you made this change can you share video of publish model to make sure it is working fine

**Comment by Yashsharma1911** [comment]
Testing

**Review by Yashsharma1911 — CHANGES_REQUESTED** [review]
Only one model showing up in publish model dropdown

Screenshot 2024-04-24 at 3 56 27 AM


instead it should show like this

Screenshot 2024-04-24 at 3 57 32 AM

**Comment by Yashsharma1911** [comment]
Make sure state is properly updating

File: ui/components/Filters.js [hunk 5]
```diff
@@ -336,44 +338,30 @@ function MesheryFilters({
...
-          setPublishSchema({ rjsfSchema: modifiedSchema, uiSchema: result.uiSchema });
-          setMeshModels(models);
-        } catch (err) {
-          console.error(err);
-          setPublishSchema(result);
-        }
-      },
-      handleError(ACTION_TYPES.SCHEMA_FETCH),
-    );
+    async function fetchMeshModels() {
```
Comment (codeKraken19) [inline ui/components/Filters.js:341]: use rte-query here ...

**Review by codeKraken19 — CHANGES_REQUESTED** [review]

**Review by betelgeuse-code — COMMENTED** [review]
Uhh ho lint check failing @11happy

**Comment by 11happy** [comment]
I will make the required changes soon.

**Comment by 11happy** [comment]
Hello @Yashsharma1911 can you take a look the technology section is working fine now, In `ui/components/MesheryPatterns.js` I am not able to remove that api code for design fetch as it is causing the issue. code causing issue:
```
dataFetch(
      '/api/schema/resource/design',
      {
        method: 'GET',
        credentials: 'include',
      },
      (result) => {
        setImportSchema(result);
      },
      handleError(ACTION_TYPES.SCHEMA_FETCH),
    );
  
```

**Review by betelgeuse-code — COMMENTED** [review]
> Hello @Yashsharma1911 can you take a look the technology section is working fine now, In `ui/components/MesheryPatterns.js` I am not able to remove that api code for design fetch as it is causing the issue. code causing issue:
> 
> ```
> dataFetch(
>       '/api/schema/resource/design',
>       {
>         method: 'GET',
>         credentials: 'include',
>       },
>       (result) => {
>         setImportSchema(result);
>       },
>       handleError(ACTION_TYPES.SCHEMA_FETCH),
>     );
>   
> ```

@11happy What issue you are getting? can you share that? you have made sure you are using schema from Sistent for import modal right?

File: ui/components/Filters.js (hunk 5 above)
Comment (betelgeuse-code) [reply ui/components/Filters.js:341]: @11happy thoughts? are your familiar with RTK queries? if not no worries let's move forward with this PR as you fix the issue you are getting

Comment (11happy) [reply ui/components/Filters.js:341]: Hello @betelgeuse-code , I am not that familiar with RTK queries I will need to learn that in order to implement rtk query.

**Comment by 11happy** [comment]
hello @betelgeuse-code , when I remove this api(above mentioned) code, in the technology section only kubernetes is shown as pointed by yash above.
image

File: ui/components/Filters.js (hunk 5 above)
Comment (betelgeuse-code) [reply ui/components/Filters.js:341]: Let's move forward here you can update later when you get familiar with RTK

File: ui/components/Filters.js [hunk 6]
```diff
@@ -336,44 +338,31 @@ function MesheryFilters({
...
+
+        // Modify the schema to include mesh models
+        const modifiedSchema = modifyRJSFSchema(
+          publishCatalogItemSchema, // Use publishSchema as the base schema
+          'properties.compatibility.items.enum',
+          modelNames,
+        );
+
+        // Set the modified schema and UI schema
+        setPublishSchema({ rjsfSchema: modifiedSchema, uiSchema: publishCatalogItemUiSchema });
```
Comment (betelgeuse-code) [inline ui/components/Filters.js:355]: @11happy please see here in this state you are having object contains two properties rjsfSchema and uiSchema which is good however checkout this line, https://github.com/meshery/meshery/pull/10720/files#diff-6ec8e150ffa5db5b0830cd1598d529bbc98c79c98df32092858a51a779fcdea0R1471 over here you are just passing state itself instead you should pass it like 

```
      schema={publishFormSchema.rjsfSchema}
      uiSchema={publishFormSchema.uiSchema}
```

File: ui/components/Filters.js [hunk 7]
```diff
@@ -1470,13 +1463,13 @@ const ImportModal = React.memo((props) => {
...
-  const { publishFormSchema, handleClose, handlePublish, title } = props;
+  const { handleClose, handlePublish, title } = props;
 
   return (
     <Modal
       open={true}
-      schema={publishFormSchema.rjsfSchema}
-      uiSchema={publishFormSchema.uiSchema}
+      schema={publishCatalogItemSchema}
+      uiSchema={publishCatalogItemSchema}
```
Comment (betelgeuse-code) [inline ui/components/Filters.js:1472]: Why this change? you are getting issue because of this please revert this change or use below code it should fix your issue.

```suggestion
      schema={publishFormSchema.rjsfSchema}
      uiSchema={publishFormSchema.uiSchema}
```

**Comment by betelgeuse-code** [comment]
@11happy lint check failing

**Comment by 11happy** [comment]
As soon as my academic commitments ease up by 8-9 may, I'll promptly update the PR. Thank you for being patient with this PR.

**Comment by 11happy** [comment]
I have fixed the merge conflict, fixed the lint error .
Thank you

**Comment by sudhanshutech** [comment]
@11happy can share latest recording what it looks like

**Comment by 11happy** [comment]
https://github.com/meshery/meshery/assets/76656712/353a58c1-65d4-47cb-a96a-1d541f496556

@sudhanshutech it is correclty showing the technologies now which earlier was not.

**Comment by 11happy** [comment]
@sudhanshutech I have corrected the models list & removed the duplicates earlier showing its working fine now.Can you please take a look.
https://github.com/meshery/meshery/assets/76656712/db192165-1ff9-4d5a-a8fb-4930e7834617

**Comment by leecalcote** [comment]
Thanks, @11happy 👍 

@sudhanshutech ready for review.

**Comment by leecalcote** [comment]
@11happy there is a merge conflict here to resolve.

**Comment by sudhanshutech** [comment]
This pr : https://github.com/meshery/meshery/pull/11463 will take care of this.
